src/lib/nextWordGuesserStrategy.ts

```diff
@@ -0,0 +1,74 @@
+import { maxBy, uniq } from 'lodash';
+import { WordGuessAndResult, WordGuessAndScore } from './wordGuessAndResult';
+import WordList from './wordList';
+import { NoMoreGuessesError } from './wordleSolverError';
+
+/**
+ * TODO: Solve styles / architecture notes to self:
+ * 
+ * Abstract class with base solver stuff;
+ * Strategy methods for score methods implemented by extenders
+ * Guesser class for each method which implements strategy, extends base class;
+ * Algorithm uses all guessers and picks which one removes most possibilities.
+ *
+ * 1. guess word by eliminating highest count of distinct unguessed letters (e.g., BREAD (5) > BREED (4) = BLEED (4) > PAPAS (3))
+ * 2. guess word by highest count of additional letter+position exclusions (total # eliminations at each position maximized);
+ * 3. guess word by highest count of re-used misplaced letters; (if known TSO__ misplaced, try ST_O_, etc.)
+ * 4. guess word by highest count by most frequent letters (calculate frequency list of each position, eliminate letters highest on that, total # = metric)
+
+ * then pick from among these which removes the most amount of words total from the candidate list.
+ */
+
```

Comment:
Move these all out into the README or similar doc?

Reply:
Requires significantly more work, and already increases scope of this PR quite a bit. Will instead be resolved via https://github.com/codergeek42/wordle-solver/issues/53.